src/researchhub_comment/related_models/rh_comment_model.py

```diff
@@ -77,23 +77,34 @@ def is_root_comment(self):
     """ --- METHODS --- """
 
     @classmethod
-    def create_from_request(cls, request, rh_thread):
-        request_data = request.data
+    def create_from_data(cls, data, rh_thread):
```

Comment:
Let's wrap this in an atomic transaction `with transaction.atomic():`, `from django.db import transaction`.

I wasn't really sure about having non retrieval db actions wrapped in an atomic block, but I think it's a good practice to do now, and it is something that I am trying to do now too.